src/test/java/com/artipie/asto/SubStorageTest.java

```diff
@@ -106,4 +106,73 @@ void checksExistence(final String pref) {
             new SubStorage(Key.ROOT, this.asto).exists(new Key.From(prefix, "any.txt")).join()
         );
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"my-project", "com/example"})
+    void savesValue(final String pref) {
```

Comment:
I'm not sure this test method name and correspond to each other. To test `save` method of the `SubStorage` seems logical to 
1) create instance of the `SubStorage` with different prefixes
2) call `save` method of these `SubStorage` instances
3) check that content is saved and the key is correct by getting the value from the original storage by prefixed key. Thus, assuming, that original storage works correctly, we can be sure that: a) content is stored with the expected key b) content is correct

Reply:
olenagerasimova Got it :)